src/TechLanches/Adapter/Driver/TechLanches.Adapter.API/Endpoints/PagamentoEndpoints.cs

```diff
@@ -30,12 +36,18 @@ private static async Task<IResult> BuscarStatusPagamentoPorPedidoId([FromRoute]
                 return Results.NotFound(new ErrorResponseDTO { MensagemErro = $"Nenhum pedido encontrado para o id: {pedidoId}", StatusCode = (int)HttpStatusCode.NotFound });
 
 
-            return Results.Ok(pagamento.Adapt<PagamentoResponseDTO>());
+            return Results.Ok(pagamento.Adapt<PagamentoStatusResponseDTO>());
         }
 
-        private static async Task<IResult> BuscarPagamento()///?????
+        private static async Task<IResult> BuscarPagamento(int mercadoPagoVendaId, [FromServices] IPagamentoQrCodeACLService pagamentoQrCodeACLService)
         {
-            throw new NotImplementedException();
+            var pagamento =  await pagamentoQrCodeACLService.ConsultarPagamento(mercadoPagoVendaId);
+
```

Comment:
o fluxo desse endpoint seria um pouco diferente, receberiamos 2 parametros do mercado pago via query string (topic, id)  e um nosso caso for fluxo local (pedidoId), precisariamos verificar se o topic é do tipo merchant_order, e ai caso for, delegariamos pra uma service que chamaria a ACL,  e ai dependendo do retorno da acl (pagamento aprovado ou reprovado) atualizariamos no banco a entidade pagamento e o status da entidade pedido (criado ou cancelado) e ai retornamos apenas Ok no endpoint, caso contrario o mercado pago retentara enviar novamente